specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json

```diff
@@ -2303,6 +2303,10 @@
     "CosmosDbMongoDbApiLinkedServiceTypeProperties": {
       "description": "CosmosDB (MongoDB API) linked service properties.",
       "properties": {
+        "isServerVersionAbove32": {
+          "type": "object",
```

Comment:
Why it's object type?

Reply:
To support parameterization, we used object type in most of properties.

Reply:
Hello @PhoenixHe-msft , could you please help to check again?

Reply:
We should avoid using a free-form object. Platform feature like Azure policy won't work with it.

[https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy005-avoid-free-form-objects](https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy005-avoid-free-form-objects)

What's the issue with setting it to string or boolean type?

Reply:
Hi @j5lim , thanks for the info, but it's a by design requirement, you can see we set most of properties to free-from object.
The free-from object is used to support parameterization, in that case customer can use an expression to calculate the property value.
Parameterization is supported in our UI, if we set a fixed type here, the payload exported from UI that using parameterization cannot load into SDK.